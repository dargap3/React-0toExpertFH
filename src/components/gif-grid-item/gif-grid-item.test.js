import { shallow } from "enzyme";

import GifGridItem from './gif-grid-item.jsx';


describe('<GifGridItem /> tests', () => {
  const title = 'Hola';
  const url = 'https://www.google.com';
  const wrapper = shallow(<GifGridItem title={ title } url={ url } />);

  test('should show <GifGridItem /> correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should have a paragraph with title', () => {
    const p = wrapper.find('p');

    expect(p.text().trim()).toBe(title);  
  });

  test('should have url and alt same as props', () => {
    const img = wrapper.find('img');

    expect(img.prop('src')).toBe(url);
    expect(img.prop('alt')).toBe(title);
  });

  test('should have animate__fadeIn className', () => {

    const div = wrapper.find('div');
    const className = div.prop('className');

    expect(className.includes('animate__fadeIn')).toBe(true);  
  });  
  
});