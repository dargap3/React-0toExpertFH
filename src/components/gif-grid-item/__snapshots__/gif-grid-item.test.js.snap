// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> tests should show <GifGridItem /> correctly 1`] = `
<div
  className="card animate__fadeIn"
>
  <img
    alt="Hola"
    loading="lazy"
    src="https://www.google.com"
  />
  <p>
    Hola
  </p>
</div>
`;
